spacy/lemmatizer.py

```diff
@@ -92,18 +92,32 @@ def lemmatize(string, index, exceptions, rules):
     forms = []
     forms.extend(exceptions.get(string, []))
     oov_forms = []
+    exp_forms = []
     if not forms:
         for old, new in rules:
             if string.endswith(old):
                 form = string[:len(string) - len(old)] + new
                 if not form:
                     pass
                 elif form in index or not form.isalpha():
+                    exp_forms.append((string[:len(string)-len(old)] , old, form))
```

Comment:
This won't probably pass the linter. You should use something like Flake8 to make sure the code complies with the style guide. That should be rewritten to:
```
exp_forms.append((string[:len(string) - len(old)], old, form))
```